### ISSUE TITLE
Multiblock factories

### ISSUE THREAD

**Issue by RoyalFoxy** [mekanism/Mekanism-Feature-Requests#560]
### Describe the feature you'd like

So I wondered why there are so many multiblocks for different things like power generators, tanks, batteries etc. but not for the machines that produce most items in Mekanism themselves like smelting factories, chemical oxidizers or a rotary condensentrator. I thought those multiblocks would be endgame after you've unlocked antimatter and would require pellets to craft them as those machines would require massive amounts of energy, input and would have an massive output thus the "stress" would be very high meaning a sort of "glue" material like antimatter would be a good option to use for them. depending on the size of the multiblock structure they could house more or less upgrade, have more slots to process input materials and have slightly higher speeds aswell.

Some things I thought of are

Hulls:
Meaning for a machine that would process radioactive material the hull would have to be crafted with lead e.g. like the fission reactor casing or the isotopic centrifuge.

Insides:
A machine should contain at least one of its dedicated processing unit like a smelting factory unit to smelt items. additional blocks like energy, speed or gas units could be added to improve their performance in their catagories by a considerable amount compared to factory unit. Lets say one smelting unit would be crafted with 4 ultimate smelting factories then that unit would have 4x the speed, energy and parallel processing power than one ultimate smelting factory. If you then add a speed unit into the machine it could add 120% to the speed so a total of 220% * 4 so 880% faster than one ultimate smelting factory. Those are just numbers I made up, I haven't looked into the balancing side of things of my idea. The same process would apply for other types of units you could add. The input and output slots should be scrollable to account for massive amounts of inputs as the machines shouldn't just hold one smelting factory unit but a couple to truely make it a big brother multiblock.

upgrades:
I thought it would only be **reasonable** if you could also put it 4x the upgrades for each unit you add so 32 speed and energy upgrades for a smelting factory unit. This would obviously require a ton of energy to power if you add 32 speed upgrades but it would also be insanely fast and you'd probably get away with just one or two of the specific units you want to use unless you're going full megabase and need to process thousands of resources at a time.

First balancing ideas for crafting:
To not make this too overpowered you could make it so that one smelting factory unit from above would require a steel casing, 4 ultimate smelting factories and 4 antimatter pellets which would make it fair in my opinion as it also enables you to process hundreds of items in parallel if you build it big and also allows you not havng to build dozens if not hundreds of machines yourself but just build one big one. For upgrade units I thought it would be a good idea to also add a few new items aswell which would be upgrade stacks. Those would be crafted with 8 upgrades surrounding a polonium pellet. With 4 of those upgrade stacks, 4 antimatter pellets and another steel casing you could craft yourself an upgrade unit which would contain 32 upgrades of its kind.

This is the general idea for a new way to process massive amounts of resources quickly and efficiently. Let me know if you think this is not suited for mekanism and I should just continue building dozens of factories or if this could eventually be added in some form or way

### Describe alternatives you've considered

I thought of something like issue [376](https://github.com/mekanism/Mekanism-Feature-Requests/issues/376) describes

**Comment by Michael-Codelabs** [comment]
I agree, And this is also needed for many other machines, like pumps, rotary condenstrators and anything that isn´t block upgradable to be honest could use that addition.
Maybe ever if done right they could mode each version to be a different size and model with each upgrade.
The same sort of modeling as a modification station.

**Comment by RoyalFoxy** [comment]
@CodeLabsCEO I think if you go that far it would be a huge update and this request could take months, maybe even years of coding, testing and balancing if almost every machine that doesn't already have a bigger multiblock structure would get one. Sorry for the late response, didn't keep an eye out for this
